frontend/src/lib/components/LemonTable/LemonTable.tsx

```diff
@@ -120,16 +99,23 @@ export function LemonTable<T extends Record<string, any>>({
     const scrollRef = useRef<HTMLDivElement>(null)
 
     /** Number of entries in total. */
-    const entryCount = pagination && 'entryCount' in pagination ? pagination.entryCount : dataSource.length
+    const entryCount: number | null = pagination?.controlled ? pagination.entryCount || null : dataSource.length
     /** Number of pages. */
-    const pageCount = pagination ? Math.ceil(entryCount / pagination.pageSize) : 1
+    const pageCount: number | null =
+        entryCount && (pagination ? (pagination.pageSize ? Math.ceil(entryCount / pagination.pageSize) : 1) : null)
     /** Page adjusted for `pageCount` possibly having gotten smaller since last page param update. */
-    const currentPage =
-        pagination && 'currentPage' in pagination
-            ? pagination.currentPage
-            : Math.min(parseInt(searchParams[currentPageParam]) || 1, pageCount)
+    const currentPage: number | null = pagination?.controlled
+        ? pagination.currentPage || null
+        : Math.min(parseInt(searchParams[currentPageParam]) || 1, pageCount as number)
```

Comment:
So the cast is there because `pageCount` CANNOT be `null` in this place, in accordance with types, it's just that TypeScript is not that smart to figure that out. Added a `// Note:` about this